src/components/Reports/PartnerGivingAnalysisReport/Table/Table.tsx

```diff
@@ -178,18 +174,20 @@ export const PartnerGivingAnalysisTable: React.FC<
       }}
     >
       <StyledDataGrid
+        apiRef={apiRef}
         rows={tableRows}
-        rowCount={totalCount}
         columns={columns}
         getRowId={(row) => row.id}
         sortingOrder={['asc', 'desc']}
+        initialState={{
+          // Set initial pagination to first page with 25 rows per page
+          pagination: {
+            paginationModel: { pageSize: 25, page: 0 },
```

Comment:
It looks like we don't have to specify the page if we want the first page

```suggestion
          // Set initial pagination to 25 rows per page
          pagination: {
            paginationModel: { pageSize: 25 },
```